sky/execution.py

```diff
@@ -226,12 +226,12 @@ def _execute(
             finally:
                 # Enables post_execute() to be run after KeyboardInterrupt.
                 backend.post_execute(handle, down)
-    finally:
+
         if stages is None or Stage.DOWN in stages:
             if down and idle_minutes_to_autostop is None:
                 backend.teardown_ephemeral_storage(task)
                 backend.teardown(handle, terminate=True)
-
+    finally:
```

Comment:
Can we document the fact that any error that happens during provisioning/data syncing/setup will ignore the --down request and preserve the cluster for debugging?  In the help strings + docstrs.

Reply:
Added. Thanks!